trace.go

```diff
@@ -144,6 +144,27 @@ func GetFields(err error) map[string]interface{} {
 	return map[string]interface{}{}
 }
 
+// UnwrapProxyField attempts to find a named field within an error returned from ReadError.
+// If found, the field is reinterpreted through json.Unmarshal call and stored in unmarshalPtr.
+// Returns true on success and false on failure.
+func UnwrapProxyField(err error, fieldName string, unmarshalPtr interface{}) bool {
+	if proxyErr, ok := err.(proxyError); ok {
```

Comment:
If you invert all `ok`'s to `!ok`'s, the method will be much easier to follow and everything will be indented left.

```go
proxyErr, ok := err.(proxyError)
if !ok {
  return false
}
proxyDeepTraceErr, ok := proxyErr.Err.(*TraceErr)
if !ok {
  return false
}
...
```